code/modules/mob/mob.dm

```diff
@@ -1182,6 +1182,8 @@ GLOBAL_LIST_INIT(slot_equipment_priority, list( \
 			if(!no_text)
 				visible_message("<span class='warning'>[src] pukes all over [p_them()]self!</span>","<span class='warning'>You puke all over yourself!</span>")
 			location.add_vomit_floor(TRUE)
+	if (nutrition > 0)
```

Comment:
Logically, I think this check could do a lot more for this proc (and add some more functionality to it).

I believe your intent of this PR is to prevent people from vomit spamming, yes this will give people hypoglycemia but it won't actually stop them from vomiting all the way until they die.

I think you could instead do this.

```
if (nutrition > 0)
    if(istype(location, /turf/simulated))
        VOMIT CODE
    adjust_nutrition(rand(-30, -15))
else
    visible_message("<span class='warning'>[src] dry heaves!</span>","<span class='warning'>You puke but there's nothing left in your stomach!</span>")
    return```

This would allow us to escape this proc quicker and remove the need for calling some procs more than we actually should.

Reply:
oh yeah that would also work preventing puking on empty stomach I'll update this in a bit

Reply:
Changed it to what you suggest also just moved the **adjust_nutrition()**'s scope so you wont lose nutrition when their are no acceptable tiles/when you're missing the message telling you

Reply:
Would be appreciated if reviewed again @Sirryan2002

Reply:
Will do, I'm omw back to uni rn so it might be s few hours